its/ruling/src/test/java/org/sonar/java/it/SonarWayTest.java

```diff
@@ -0,0 +1,117 @@
+    String projectName = "Java Checks Test Sources";
+
+    orchestrator.getServer().provisionProject(projectKey, projectName);
+    orchestrator.getServer().associateProjectToQualityProfile(projectKey, "java", "rules");
+
+    /**
+     * 1. Run the analysis as maven project
+     */
+    String correctConfigIssues = FileLocation.of("target/actual/" + projectKey + "-mvn").getFile().getAbsolutePath();
+
+    MavenBuild mavenBuild = MavenBuild.create()
+      .setPom(FileLocation.of(projectLocation + "pom.xml").getFile().getCanonicalFile())
+      .setCleanPackageSonarGoals()
+      .addArgument("-DskipTests")
+      .addArgument("-Panalyze-tests")
+      .setProperty("sonar.projectKey", projectKey)
+      .setProperty("sonar.projectName", projectName)
+      .setProperty("sonar.cpd.exclusions", "**/*")
+      .setProperty("sonar.skipPackageDesign", "true")
+      .setProperty("sonar.lits.dump.old", TMP_DUMP_OLD_FOLDER.newFolder().getAbsolutePath())
+      .setProperty("sonar.lits.dump.new", correctConfigIssues)
+      .setProperty("sonar.lits.differences", FileLocation.of("target/" + projectKey + "-mvn_differences").getFile().getAbsolutePath())
+      .setProperty("sonar.internal.analysis.failFast", "true");
+
+    orchestrator.executeBuild(mavenBuild);
+
+    /**
+     * 2. execute the analysis as sonar-scanner project, without any bytecode
+     */
+    SonarScanner sonarScannerBuild = SonarScanner.create(FileLocation.of(projectLocation).getFile())
+      .setProjectKey(projectKey)
+      .setProjectName(projectName)
+      .setProjectVersion("0.1.0-SNAPSHOT")
+      .setSourceEncoding("UTF-8")
+      .setSourceDirs("src/main/java/")
+      .setTestDirs("src/test/java/")
+      .setProperty("sonar.cpd.exclusions", "**/*")
+      .setProperty("sonar.skipPackageDesign", "true")
+      .setProperty("sonar.java.source", "17")
+      // Dummy sonar.java.binaries to pass validation
+      .setProperty("sonar.java.binaries", TMP_DUMP_OLD_FOLDER.newFolder().getAbsolutePath())
+      // use as "old" issues the ones from Maven analysis
+      .setProperty("sonar.lits.dump.old", correctConfigIssues)
+      .setProperty("sonar.lits.dump.new", FileLocation.of("target/actual/" + projectKey + "-no-binaries").getFile().getAbsolutePath())
+      .setProperty("sonar.lits.differences", FileLocation.of("target/" + projectKey + "-no-binaries_differences").getFile().getAbsolutePath())
+      .setProperty("sonar.internal.analysis.failFast", "true");
+
+    orchestrator.executeBuild(sonarScannerBuild);
+
+    // 5min sleep
+    LOG.info("WAITING 5MIN BEFORE COMPARING RESULTS (SERVER: " + orchestrator.getServer().getUrl() + ")");
+    Thread.sleep(1000 * 60 * 5);
+
+    /**
+     * 3. Check differences
+     */
+    String differences = new String(Files.readAllBytes(FileLocation.of("target/" + projectKey + "-no-binaries_differences").getFile().toPath()), StandardCharsets.UTF_8);
+    Assertions.assertThat(differences).isEmpty();
```

Comment:
This is definitely failing for the time being. What I would like to check is that there is no "new issues", but only "missing" ones.